ionoscloud/resource_ipfailover.go

```diff
@@ -212,57 +247,59 @@ func resourceLanIPFailoverDelete(ctx context.Context, d *schema.ResourceData, me
+	if len(parts) != 3 || parts[0] == "" || parts[1] == "" || parts[2] == "" {
+		return nil, fmt.Errorf("invalid import ID: %s. Expecting {datacenter}/{lan}/{ip}", d.Id())
 	}
-
 	dcId := parts[0]
 	lanId := parts[1]
+	ip := parts[2]
 
 	client := meta.(services.SdkBundle).CloudApiClient
 
 	lan, apiResponse, err := client.LANsApi.DatacentersLansFindById(ctx, dcId, lanId).Execute()
-	logApiRequestTime(apiResponse)
+	apiResponse.LogInfo()
 
 	if err != nil {
-		if apiResponse != nil && apiResponse.Response != nil && apiResponse.StatusCode == 404 {
+		if apiResponse.HttpNotFound() {
 			d.SetId("")
-			return nil, fmt.Errorf("an error occured while trying to fetch the lan %q", lanId)
+			return nil, fmt.Errorf("unable to find the LAN with ID: %s, datacenter ID: %s", lanId, dcId)
 		}
-		return nil, fmt.Errorf("lan does not exist %q", lanId)
+		return nil, fmt.Errorf("error while fetching LAN with ID: %s, datacenter ID: %s, err: %w", lanId, dcId, err)
 	}
 
 	log.Printf("[INFO] lan found: %+v", lan)
 
-	d.SetId(*lan.Id)
-
-	if err := d.Set("datacenter_id", dcId); err != nil {
-		return nil, err
-	}
-
-	failoverSlice := lan.Properties.IpFailover
-	if lan.Properties != nil && failoverSlice != nil && len(*failoverSlice) > 0 {
-		firstFailover := (*failoverSlice)[0]
-		if firstFailover.Ip != nil {
-			err := d.Set("ip", firstFailover.Ip)
-			if err != nil {
-				return nil, fmt.Errorf("error while setting ip property for IpFailover %s: %w", d.Id(), err)
-
-			}
-		}
-		if firstFailover.NicUuid != nil {
-			err := d.Set("nicuuid", firstFailover.NicUuid)
-			if err != nil {
-				return nil, fmt.Errorf("error while setting nicuuid property for IpFailover %s: %w", d.Id(), err)
+	ipFailoverGroups := lan.Properties.IpFailover
+	ipFailoverGroupFound := false
+	if lan.Properties != nil && ipFailoverGroups != nil && len(*ipFailoverGroups) > 0 {
+		for _, ipFailoverGroup := range *ipFailoverGroups {
+			// Search for the appropiate IP Failover Group using the provided IP
+			if *ipFailoverGroup.Ip == ip {
+				// Set all the information only if the IP Failover Group exists
+				// Use the IP in order to generate the resource ID
+				d.SetId(uuid.NewV5(uuid.NewV5(uuid.NamespaceURL, "https://github.com/ionos-cloud/terraform-provider-ionoscloud"), ip).String())
+
+				if err := d.Set("datacenter_id", dcId); err != nil {
+					return nil, utils.GenerateSetError(constant.ResourceIpFailover, "datacenter_id", err)
+				}
+				if err := d.Set("lan_id", lanId); err != nil {
+					return nil, utils.GenerateSetError(constant.ResourceIpFailover, "lan_id", err)
+				}
+				if err := d.Set("ip", ip); err != nil {
+					return nil, utils.GenerateSetError(constant.ResourceIpFailover, "ip", err)
+				}
+				if err := d.Set("nicuuid", *ipFailoverGroup.NicUuid); err != nil {
+					return nil, utils.GenerateSetError(constant.ResourceIpFailover, "nicuuid", err)
+				}
+				ipFailoverGroupFound = true
```

Comment:
can't you just return []*schema.ResourceData{d}, nil here and return the error in all other cases?

Reply:
@cristiGuranIonos done in ed1e170620e3d94cb90595dc3047a47258d4ae99